File: 평강/app/project/src/main/java/com/eventpurchase/project/domain/product/entity/ProductEntity.java
package com.eventpurchase.project.domain.product.entity;

import com.eventpurchase.project.global.common.entity.BaseTimeEntity;
import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;


@Entity
@Getter
@Builder
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor
@Table(name = "products")
public class ProductEntity extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String productName;

    @Column(nullable = false)
    private String brand;

    @Column(nullable = false)
    private String category;

    @Column
    private String subCategory;

    @Column(nullable = false)
    private BigDecimal price;

    private BigDecimal discountRate;
    private BigDecimal rating;
    private Integer reviewCount;

    @Column(columnDefinition = "TEXT")
    private String imageUrl;

    @Column(columnDefinition = "TEXT")
    private String sourceUrl;

    private String source;


}

File: 평강/app/project/product/repository-jpa/src/main/kotlin/com/eventpurchase/project/product/ProductEntity.kt
package com.eventpurchase.project.product

import com.eventpurchase.project.shared.BaseTimeEntity
import jakarta.persistence.*
import java.math.BigDecimal

@Entity
@Table(name = "products")
class ProductEntity(
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,
    @Column(nullable = false) val productName: String,
    @Column(nullable = false) val brand: String,
    @Column(nullable = false) val category: String,
    val subCategory: String? = null,
    @Column(nullable = false) val price: BigDecimal,
    val discountRate: BigDecimal? = null,
    val rating: BigDecimal? = null,
    val reviewCount: Int? = null,
    @Column(length = 1000) val imageUrl: String? = null,
    @Column(length = 1000) val sourceUrl: String? = null,
    val source: String? = null
) : BaseTimeEntity() {

    fun toDomain() = Product(id, productName, brand, category, subCategory, price, discountRate, rating, reviewCount, imageUrl, sourceUrl, source)

    companion object {
        fun from(p: Product) = ProductEntity(p.id, p.productName, p.brand, p.category, p.subCategory, p.price, p.discountRate, p.rating, p.reviewCount, p.imageUrl, p.sourceUrl, p.source)
    }
}
